src/app/core/services/phase.service.ts

```diff
@@ -82,10 +83,8 @@ export class PhaseService {
   storeSessionData(): Observable<boolean> {
     return this.fetchSessionData().pipe(
       assertSessionDataIntegrity(),
-      map((sessionData: SessionData) => {
-        this.updateSessionParameters(sessionData);
-        return this.currentPhase !== undefined;
-      })
+      tap(this.updateSessionParameters),
```

Comment:
It seems like when you pass a function to the `tap` operator, the following error occurs.

![image](https://user-images.githubusercontent.com/22557857/96060633-a1bf1c00-0ec3-11eb-8ad0-f2a4768d5ca2.png)


During the execution of `updateSessionParameters` function, the `this` will be binded to `TapSubscriber` class instead. 

Maybe we can try something like this instead?

```
tap((data: SessionData) => this.updateSessionParameters(data)),
```